### ISSUE TITLE
Connecting error not being immediately caught

### ISSUE THREAD

**Issue by mathsalmi** [typeorm/typeorm#264]
Consider this:

```typescript
orm.useContainer(Container);
orm.createConnection().then(async conn => {
	console.log('Success!');
	await conn.entityManager.query(`show tables`);
}).catch(err => console.error('Failed to start up server: ', err));
```

When I put the right credentials on ormconfig.json, everything works fine. But with the wrong credentials, I get this:

```
Success!
Failed to start up server:  { Error: getaddrinfo ENOTFOUND localhosxt localhosxt:3306
    at errnoException (dns.js:28:10)
    at GetAddrInfoReqWrap.onlookup [as oncomplete] (dns.js:76:26)
    --------------------
    at Protocol._enqueue (/Users/salmi/Downloads/teste/node_modules/mysql/lib/protocol/Protocol.js:141:48)
    at Protocol.handshake (/Users/salmi/Downloads/teste/node_modules/mysql/lib/protocol/Protocol.js:52:41)
    at PoolConnection.connect (/Users/salmi/Downloads/teste/node_modules/mysql/lib/Connection.js:130:18)
    at Pool.getConnection (/Users/salmi/Downloads/teste/node_modules/mysql/lib/Pool.js:48:16)
    at /Users/salmi/Downloads/teste/node_modules/typeorm/driver/mysql/MysqlDriver.js:248:28
    at MysqlDriver.retrieveDatabaseConnection (/Users/salmi/Downloads/teste/node_modules/typeorm/driver/mysql/MysqlDriver.js:247:20)
    at MysqlDriver.<anonymous> (/Users/salmi/Downloads/teste/node_modules/typeorm/driver/mysql/MysqlDriver.js:140:51)
    at step (/Users/salmi/Downloads/teste/node_modules/typeorm/driver/mysql/MysqlDriver.js:32:23)
    at Object.next (/Users/salmi/Downloads/teste/node_modules/typeorm/driver/mysql/MysqlDriver.js:13:53)
    at /Users/salmi/Downloads/teste/node_modules/typeorm/driver/mysql/MysqlDriver.js:7:71
  code: 'ENOTFOUND',
  errno: 'ENOTFOUND',
  syscall: 'getaddrinfo',
  hostname: 'localhosxt',
  host: 'localhosxt',
  port: 3306,
  fatal: true }
```

Shouldn't it go directly to the catch() function instead of going to the then()?

**Comment by joffray589** [comment]
I think this issue has been resolved in #610

**Comment by pleerock** [comment]
yeah it should be so
